plugins/imfile/imfile.c

```diff
@@ -702,27 +714,24 @@ detect_updates(fs_edge_t *const edge)
 {
 	act_obj_t *act;
 	struct stat fileInfo;
+	int restart = 0;
 
-	for(act = edge->active ; act != NULL ; ) {
+	for(act = edge->active ; act != NULL ; act = act->next) {
```

Comment:
3: act in act->next has been freed in line 724 (see comment below).

Reply:
Yes, however, if I unlink it I break the cycle and the processing begins again from the start.

Reply:
but you actually unlink it in 724?

Reply:
The problem I see when analyzing the result seems to be that clang seems to ignore [this line ](https://github.com/rsyslog/rsyslog/pull/2749/files#diff-b563a9e8d7f28e380e7fd4dc4467d3e8R969) line which guards all 3 cases it reports.

Reply:
the problem is not act->name but act itself.